### ISSUE TITLE
[Android] Slow encryption/decryption

### ISSUE THREAD

**Issue by ixje** [ixje/neon-wallet-react-native#1]
Currently key derivation by use of the `scrypt()` function is extremely slow on Android (only tested on emulator, as I don't have a real device). The same function takes ~3-4 seconds on the iOS emulator dev mode (with remote debugging disabled).

Things I've tried/checked:
- Run in production mode vs dev mode (no difference)
- Checked that enough memory is assigned to the emulator (1.5GB)
- Disabled any console.log() (no difference)
- Turned off redux-logger (no difference)
- Running with `Remote JS Debugging on`, it's fast again. But that's because execution get's off loaded to the browser ([read explanation here](https://stackoverflow.com/a/33577891/3869515)

We can't change the number of iterations (16384 to e.g. 1) because then keys created by other wallets will fail to verify. It's also less secure.

**Comment by cjoshi85** [comment]
Yes, I am facing the same issue while running it on real device.It is taking around 10 minutes while doing the encrypt.

**Comment by ixje** [comment]
thank you for testing the android side!
